### PR TITLE
Various Zoul-related fixes

### PR DESCRIPTION
**Pull request by g-oikonomou**
This pull fixes a bunch of compile errors for the Zoul platform:

* Firstly, we fix `platform.c` when the RTCC is enabled
* We fix various problems under `examples/platform-specific/zoul`
* We add travis tests for all Zoul-specific examples

Discovered thanks to work in #311!

### CODE REVIEW

**Comment by g-oikonomou** [comment]
For examples that are only meant to be used for specific Zoul boards, we update `Makefile.target` to specify them.

File: examples/platform-specific/zoul/orion/client/Makefile.target [hunk 1]
```diff
@@ -1 +1,2 @@
 TARGET = zoul
+BOARD = orion
```
Comment (simonduq) [inline examples/platform-specific/zoul/orion/client/Makefile.target]: I'm not sure the client assumes Orion

File: examples/platform-specific/zoul/orion/ip64-router/Makefile.target [hunk 2]
```diff
@@ -1 +1,2 @@
 TARGET = zoul
+BOARD = orion
```
Comment (simonduq) [inline examples/platform-specific/zoul/orion/ip64-router/Makefile.target]: yes but we also want BOARDS_ONLY, right?

**Comment by atiselsts** [comment]
What about the `LPM_CONF_MAX_PM 1` change - should we do it? I don't see it here, so I'm confused.

**Pull request closed without merging**